docs/docsite/rst/network/user_guide/network_best_practices_2.5.rst

```diff
@@ -338,7 +318,66 @@ You can also look at the backup files:
    find /tmp/backups
 
 
-If `ansible-playbook` fails, please follow the debug steps in :doc:`network_debug_troubleshooting`.
+If `ansible-playbook` fails, please follow the debug steps in :ref:`network_debug_troubleshooting`.
+
+
+.. _network-agnostic-examples:
+
+Example 2: simplifying playbooks with network agnostic modules
+==============================================================
+
+If you have two or more network platforms in your environment, you can use the network agnostic modules to simplify your playbooks. You can use network agnostic modules such as ``cli_command`` or ``cli_config`` in place of the platform-specific modules such as ``eos_config``, ``ios_config``, and ``junos_config``. This reduces the number of tasks and conditionals you need in your playbooks.
+
+.. note::
+  Network agnostic modules require the :ref:`network_cli <network_cli_connection>` connection plugin.
+
+
+Sample playbook with platform-specific modules
+----------------------------------------------
+
+This example assumes three platforms, Arista EOS, Cisco NXOS, and Juniper JunOS.  Without the network agnostic modules, a sample playbook might contain the following three tasks:
+
+.. code-block:: yaml
+
+  ---
+  - name: Run Arista command
+    eos_command:
+      commands: show ip int br
+    when: ansible_network_os == 'eos'
+
+  - name: Run Cisco NXOS command
+    nxos_command:
+      commands: show ip int br
+    when: ansible_network_os == 'nxos'
+
+  - name: Run Junos command
+    junos_command:
+      commands: show interface terse
+    when: ansible_network_os == 'junos'
+
+Simplified playbook with ``cli_command`` network agnostic module
+----------------------------------------------------------------
+
+If you replace these platform-specific modules with the network agnostic ``cli_command`` module, this playbook can be simplified to:
+
+.. code-block:: yaml
+
+  ---
+  - name: Run command and print to terminal window
+    hosts: routers
+    gather_facts: false
+
+    tasks:
+      - name: Run show command
```

Comment:
Probably can also add an example to work with multiple prompts. Since currently, only `cli_command` support multi prompt scenario will be good to have in the doc

Reply:
@ganeshrn can you point me to an example of this that I can add here?

Reply:
@ganeshrn ok found the example in the `cli_command` module examples so I'll add a pointer to that from here rather than repeat it.

Reply:
Here is an example for changing the password on a Juniper Junos device->

```
- name: CHANGE PASSWORD TO WORKSHOP DEFAULT
  cli_command:
    command: "{{item}}"
    prompt:
      - "New password"
      - "Retype new password"
    answer:
      - "mypassword123"
      - "mypassword123"
    check_all: True
  loop:
    - "configure"
    - "rollback"
    - "set system root-authentication plain-text-password"
    - "commit"
```

Reply:
added. thanks!